tests/test_regressor/test_influence.py

```diff
@@ -0,0 +1,169 @@
+    """
+    X, y = make_regression(
+        n_samples=100,
+        n_features=14,
+        n_informative=6,
+        bias=1.2,
+        noise=49.8,
+        tail_strength=0.6,
+        random_state=637,
+    )
+
+    request.cls.data = TestDataset(X, y)
+
+
+##########################################################################
+## Assertion Helpers
+##########################################################################
+
+LEARNED_FIELDS = (
+    'distance_', 'p_values_', 'influence_threshold_', 'outlier_percentage_'
+)
+
+
+def assert_not_fitted(oz):
+    for field in LEARNED_FIELDS:
+        assert not hasattr(oz, field)
+
+
+def assert_fitted(oz):
+    for field in LEARNED_FIELDS:
+        assert hasattr(oz, field)
+
+
+##########################################################################
+## Test CooksDistance Visualizer
+##########################################################################
+
+
+@pytest.mark.usefixtures("data")
+class TestCooksDistance(VisualTestCase):
+    """
+    CooksDistance visual test cases
+    """
+
+    def test_cooks_distance(self):
+        """
+        Test image similarity of Cook's Distance on a random dataset
+        """
+        _, ax = plt.subplots()
+        viz = CooksDistance(ax=ax)
+
+        assert_not_fitted(viz)
+        assert viz.fit(self.data.X, self.data.y) is viz
+        assert_fitted(viz)
+
+        # Test fitted values
+        assert viz.distance_.shape == (self.data.X.shape[0],)
+        assert viz.p_values_.shape == viz.distance_.shape
+        assert 0.0 <= viz.influence_threshold_ <= 4.0
+        assert 0.0 <= viz.outlier_percentage_ <= 100.0
+
+        self.assert_images_similar(viz)
+
+    def test_cooks_distance_quickmethod(self):
+        """
+        Test the cooks_distance quick method on a random dataset
+        """
+        _, ax = plt.subplots()
+        viz = cooks_distance(
+            self.data.X,
+            self.data.y,
+            ax=ax,
+            draw_threshold=False,
+            linefmt="r-",
+            markerfmt="ro",
+        )
+
+        assert_fitted(viz)
+        self.assert_images_similar(viz)
+
+    @pytest.mark.skipif(pd is None, reason="test requires pandas")
+    def test_pandas_integration(self):
+        """
+        Test on the concrete dataset with pandas DataFrame and Series
+        """
+        data = load_concrete(return_dataset=True)
+        X, y = data.to_pandas()
+
+        assert isinstance(X, pd.DataFrame)
+        assert isinstance(y, pd.Series)
+
+        _, ax = plt.subplots()
+        viz = CooksDistance(ax=ax).fit(X, y)
+        assert_fitted(viz)
+
+        assert viz.distance_.sum() == pytest.approx(1.2911900571300652)
+        assert viz.p_values_.sum() == pytest.approx(1029.9999525376425)
+        assert viz.influence_threshold_ == pytest.approx(0.003883495145631068)
+        assert viz.outlier_percentage_ == pytest.approx(7.3786407766990285)
```

Comment:
Because this is a static data set we can test real values. Note that `viz.distance_.sum()` is an approximation for the correct values; if for some reason `X` is sorted this test will still pass even though `distance_` is incorrect. I think it is worth the trade-off to have readability in tests and coverage for 90% of cases than to test an exact dataset!